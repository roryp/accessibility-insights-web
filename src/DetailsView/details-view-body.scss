// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../common/styles/common.scss';

.details-view-body {
    height: 100%;
}

.details-view-content-pane-container {
    display: flex;
    flex-direction: column;
    height: 100%;
    width: 100%;
    padding-bottom: 1px;
    box-sizing: border-box;
    max-height: calc(100vh - (#{$detailsViewTotalHeaderHeight}));
    &:global(.reflow-ui) {
        width: calc(100vw - (#{$detailsViewReflowLeftNavWidth}));
        max-height: calc(100vh - (#{$detailsViewHeaderBarHeight}));
    }
    &.narrow-mode {
        width: 100%;
    }
}
