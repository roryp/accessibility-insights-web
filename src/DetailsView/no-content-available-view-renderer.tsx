// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { DocumentManipulator } from 'common/document-manipulator';
import {
    NoContentAvailableView,
    NoContentAvailableViewDeps,
} from 'DetailsView/components/no-content-available/no-content-available-view';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { config } from '../common/configuration';

export class NoContentAvailableViewRenderer {
    constructor(
        private readonly deps: NoContentAvailableViewDeps,
        private readonly dom: Document,
        private readonly renderer: typeof ReactDOM.render,
        private readonly documentManipulator: DocumentManipulator,
    ) {}

    public render(): void {
        const detailsViewContainer = this.dom.querySelector('#details-container');
        const iconPath = '../' + config.getOption('icon128');
        this.documentManipulator.setShortcutIcon(iconPath);

        this.renderer(<NoContentAvailableView deps={this.deps} />, detailsViewContainer);
    }
}
