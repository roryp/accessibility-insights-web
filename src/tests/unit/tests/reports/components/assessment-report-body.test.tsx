// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';
import {
    AssessmentReportBody,
    AssessmentReportBodyDeps,
    AssessmentReportBodyProps,
} from 'reports/components/assessment-report-body';
import { AssessmentReportBuilderTestHelper } from '../../DetailsView/assessment-report-builder-test-helper';

describe('AssessmentReportBody', () => {
    test('render', () => {
        const deps: AssessmentReportBodyDeps = {
            outcomeTypeSemanticsFromTestStatus: {
                stub: 'outcomeTypeSemanticsFromTestStatus',
            } as any,
        } as AssessmentReportBodyDeps;

        const props: AssessmentReportBodyProps = {
            deps: deps,
            data: AssessmentReportBuilderTestHelper.getAssessmentReportModel(),
            description: 'test-description',
        };

        const wrapper = shallow(<AssessmentReportBody {...props} />);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
