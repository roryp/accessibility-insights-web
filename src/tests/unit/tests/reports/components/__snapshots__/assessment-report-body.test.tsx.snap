// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentReportBody render 1`] = `
<div
  className="assessment-report-body"
  role="main"
>
  <AssessmentReportBodyHeader />
  <AssessmentReportSummary
    summary={
      Object {
        "byPercentage": Object {
          "fail": 50,
          "incomplete": 37,
          "pass": 13,
        },
        "byRequirement": Object {
          "fail": 3,
          "incomplete": 2,
          "pass": 1,
        },
        "reportSummaryDetailsData": Array [
          Object {
            "displayName": "Assessment1",
            "fail": 2,
            "incomplete": 1,
            "pass": 1,
          },
          Object {
            "displayName": "Assessment2",
            "fail": 1,
            "incomplete": 1,
            "pass": 0,
          },
        ],
      }
    }
  />
  <AssessmentScanDetails
    description="test-description"
    details={
      Object {
        "reportDate": 2000-01-01T00:00:00.000Z,
        "targetPage": "title",
        "url": "url",
      }
    }
  />
  <div
    className="details-section"
  >
    <h2
      className="details-section-header"
    >
      Failed tests
      <OutcomeChip
        count={3}
        outcomeType="fail"
      />
    </h2>
    <AssessmentReportAssessmentList
      assessments={
        Array [
          Object {
            "displayName": "Assessment1",
            "key": "assessment1",
            "steps": Array [
              Object {
                "defaultMessageComponent": null,
                "header": Object {
                  "description": null,
                  "displayName": "Step2A",
                  "guidanceLinks": Array [],
                  "requirementType": "assisted",
                },
                "instances": Array [
                  Object {
                    "props": Array [
                      Object {
                        "key": "Path",
                        "value": "target1",
                      },
                      Object {
                        "key": "Snippet",
                        "value": "someHtml",
                      },
                    ],
                  },
                ],
                "key": "step2a",
                "showPassingInstances": false,
              },
              Object {
                "defaultMessageComponent": null,
                "header": Object {
                  "description": null,
                  "displayName": "Step4A",
                  "guidanceLinks": Array [],
                  "requirementType": "manual",
                },
                "instances": Array [
                  Object {
                    "props": Array [
                      Object {
                        "key": "Comment",
                        "value": "comment",
                      },
                    ],
                  },
                ],
                "key": "step4a",
                "showPassingInstances": false,
              },
            ],
          },
          Object {
            "displayName": "Assessment2",
            "key": "assessment2",
            "steps": Array [
              Object {
                "defaultMessageComponent": null,
                "header": Object {
                  "description": null,
                  "displayName": "Step2B",
                  "guidanceLinks": Array [],
                  "requirementType": "manual",
                },
                "instances": Array [],
                "key": "step2b",
                "showPassingInstances": true,
              },
            ],
          },
        ]
      }
      deps={
        Object {
          "outcomeTypeSemanticsFromTestStatus": Object {
            "stub": "outcomeTypeSemanticsFromTestStatus",
          },
        }
      }
      status={2}
    />
  </div>
  <div
    className="details-section"
  >
    <h2
      className="details-section-header"
    >
      Incomplete tests
      <OutcomeChip
        count={2}
        outcomeType="incomplete"
      />
    </h2>
    <AssessmentReportAssessmentList
      assessments={
        Array [
          Object {
            "displayName": "Assessment1",
            "key": "assessment1",
            "steps": Array [
              Object {
                "defaultMessageComponent": null,
                "header": Object {
                  "description": null,
                  "displayName": "Step3A",
                  "guidanceLinks": Array [],
                  "requirementType": "assisted",
                },
                "instances": Array [
                  Object {
                    "props": Array [
                      Object {
                        "key": "Path",
                        "value": "target2",
                      },
                      Object {
                        "key": "Snippet",
                        "value": "someHtml",
                      },
                    ],
                  },
                ],
                "key": "step3a",
                "showPassingInstances": false,
              },
            ],
          },
          Object {
            "displayName": "Assessment2",
            "key": "assessment2",
            "steps": Array [
              Object {
                "defaultMessageComponent": null,
                "header": Object {
                  "description": null,
                  "displayName": "Step1B",
                  "guidanceLinks": Array [],
                  "requirementType": "assisted",
                },
                "instances": Array [],
                "key": "step1b",
                "showPassingInstances": true,
              },
            ],
          },
        ]
      }
      deps={
        Object {
          "outcomeTypeSemanticsFromTestStatus": Object {
            "stub": "outcomeTypeSemanticsFromTestStatus",
          },
        }
      }
      status={1}
    />
  </div>
  <div
    className="details-section"
  >
    <h2
      className="details-section-header"
    >
      Passed tests
      <OutcomeChip
        count={1}
        outcomeType="pass"
      />
    </h2>
    <AssessmentReportAssessmentList
      assessments={
        Array [
          Object {
            "displayName": "Assessment1",
            "key": "assessment1",
            "steps": Array [
              Object {
                "defaultMessageComponent": null,
                "header": Object {
                  "description": null,
                  "displayName": "Step1A",
                  "guidanceLinks": Array [
                    Object {
                      "href": "link1A",
                      "text": "link1A",
                    },
                  ],
                  "requirementType": "assisted",
                },
                "instances": Array [
                  Object {
                    "props": Array [
                      Object {
                        "key": "Path",
                        "value": "target1",
                      },
                      Object {
                        "key": "Snippet",
                        "value": "someHtml",
                      },
                    ],
                  },
                ],
                "key": "step1a",
                "showPassingInstances": false,
              },
            ],
          },
        ]
      }
      deps={
        Object {
          "outcomeTypeSemanticsFromTestStatus": Object {
            "stub": "outcomeTypeSemanticsFromTestStatus",
          },
        }
      }
      status={0}
    />
  </div>
</div>
`;
